examples-gallery/plot_sliding_block.py

```diff
@@ -0,0 +1,333 @@
+# %%
+""""
+Sliding a Block on a Road
+=========================
+A block, modeled as a particle is sliding on a road to cross a
+hill. The block is subject to gravity and speed dependent
+friction.
+Gravity points in the negative Y direction.
+A force tangential to the road is applied to the block.
+Three objective functions to me minimized may be selected:
+
+- selektion = 0: time to reach the end point is minimized
+- selektion = 1: time to reach the end point and the energy are minimized.
```

Comment:
This is ill-defined, no? If you try to minimize time and minimize energy, those are at odds, so it would only depend on what you weight more than the other, right?

Reply:
YES, in line 109 I define the variable verstaerkung = 2.e5. (The numerical value I picked by trial and error to make it look nice) This is the relative weight of the 'time' vs. energy.
(Verstaerkung = weight or strength in German. I often use German words, to make sure I do not accidentally overwrite Python key words)